rules/metrics_collector.go

```diff
@@ -19,16 +19,25 @@ type metricsInfo struct {
 	startTime time.Time
 }
 
-func newMetricsInfo(ctx context.Context, keyPattern string) metricsInfo {
+func newMetricsInfo(ctx context.Context, keyPattern string, startTime time.Time) metricsInfo {
+	return metricsInfo{
+		keyPattern: keyPattern,
+		method:     getMethodNameFromContext(ctx),
+		startTime:  startTime,
+	}
+}
+
+func getMethodNameFromProvider(cp ContextProvider) string {
```

Comment:
Was the method name not previously set or was it populated as rules instead of the provider?

Reply:
It was obtained from the context previously, same as now. Only difference is that I moved where the context was obtained so that the timeout was correct, so the way the name was obtained had to be adjusted slightly